### DISCUSSION TITLE
3.2.0 RC1

### DISCUSSION THREAD

**Discussion by sfmskywalker** [elsa-workflows/elsa-core#5569] [Design & tech]
## What's Changed
* LogRecord storage at differents levels by @jdevillard in https://github.com/elsa-workflows/elsa-core/pull/4911
* Minor improvements and bug fixes following the 3.1 release by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5168
* Patch 3.1.2 by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5172
* Add settings for MassTransit message handling by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5176
* Improved the returned count of cancelled workflows by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5181
* Enhance BulkDispatchWorkflows Activity with Support for Multiple Input Keys and Customizable Input Key by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5182
* Update method call in BulkDispatchWorkflows by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5185
* Changed behaviour on publishing workflows by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5208
* Add caching to workflow runtime and workflow management stores by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5174
* Replace ScheduleOutcomesAsync with CompleteActivityWithOutcomesAsync by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5212
* Refactor workflow dispatch and instance creation process by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5213
* Fix LogPersistenceMode default mapping logic by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5218
* Bulk BulkDispatchWorkflows fixes by @MariusVuscanNx in https://github.com/elsa-workflows/elsa-core/pull/5221
* Fix WorkflowActivity to Use Cached Workflow Definitions for Consistent Behavior by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5223
* Optimize Workflow Execution and Messaging by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5243
* issue 5113: Allows string support for HttpFileResponse activity by @DelarreAlexis in https://github.com/elsa-workflows/elsa-core/pull/5113
* Update activity execution mapping in workflow runtime by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5253
* Cache JavaScript Expressions by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5273
* Refactor serializer options usage and enhance caching in workflow serialization by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5275
* Remove SerializedOutputValue from JournalData by @jdevillard in https://github.com/elsa-workflows/elsa-core/pull/5266
* Clone the Serializer Options to customize for specific context by @jdevillard in https://github.com/elsa-workflows/elsa-core/pull/5278
* Reuse JsonSerializerOptions by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5283
* Fix the property SerializedValue in RuntimeDbContext for Oracle by @b0rv3g4r4 in https://github.com/elsa-workflows/elsa-core/pull/5231
* Feature/sonar cloud by @cristinamudura in https://github.com/elsa-workflows/elsa-core/pull/5224
* Add Component Testing Framework by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5261
* feat(liquid): delete json filter and add filters already provided by … by @jeanbaptistedalle in https://github.com/elsa-workflows/elsa-core/pull/5188
* Update packages by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5286
* Fix Race Condition in WorkflowDefinitionActivity by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5315
* Fix FlowJoin Activity NRE Bug by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5349
* Remove CachingBookmarkStore class and its references by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5354
* Include dropdown when using a nullable enum in an activity input by @nickrubenstein in https://github.com/elsa-workflows/elsa-core/pull/5320
* Bugfixes for heartbeats using MongoDB by @stephanmelzer in https://github.com/elsa-workflows/elsa-core/pull/5337
* Add Fault Code, Category and Type by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5362
* Update Npgsql versions by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5367
* Add distributed locking to MassTransitWorkflowDispatcher by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5418
* Update Npgsql package versions by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5426
* Revert sending input as global state instead of parameter by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5425
* Global read-only and read-only workflows by @MariusVuscanNx in https://github.com/elsa-workflows/elsa-core/pull/5406
* Add Http Response Behavior to answer synchronously using the HttpResponse Activity by @jdevillard in https://github.com/elsa-workflows/elsa-core/pull/5446
* add text/html content parser as string content by @jdevillard in https://github.com/elsa-workflows/elsa-core/pull/5460
* Refactor `DefaultTriggerScheduler` and handle past `StartAt` triggers by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5463
* Update and optimize WorkflowDefinitionStore logic by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5462
* Add JSON Serialization for Elsa expression by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5490
* Remove signal capturing phase by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5501
* Add activity existence check in Flowchart by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5544
* Publishing workflows always checks for consuming workflows to update by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5535
* Refactored handling/updating of the ActivityRegistry by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5468
* Added MongoDB implementation for alteration storage by @rosca-sabina in https://github.com/elsa-workflows/elsa-core/pull/5555
* Fixed DI registration for MongoAlterationPlanStore by @rosca-sabina in https://github.com/elsa-workflows/elsa-core/pull/5564
* Add application roles and configure them in MassTransit by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5561
* Fix workflow variable scope inconsistency by @sfmskywalker in https://github.com/elsa-workflows/elsa-core/pull/5558
* Update packages to latest versions by @raymonddenhaan in https://github.com/elsa-workflows/elsa-core/pull/5567

## New Contributors
* @DelarreAlexis made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/5113
* @nickrubenstein made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/5320
* @stephanmelzer made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/5337
* @rosca-sabina made their first contribution in https://github.com/elsa-workflows/elsa-core/pull/5555

**Full Changelog**: https://github.com/elsa-workflows/elsa-core/compare/3.1.2...3.2.0-rc1

<hr /><em>This discussion was created from the release <a href='https://github.com/elsa-workflows/elsa-core/releases/tag/3.2.0-rc1'>3.2.0 RC1</a>.</em>

**Comment by glime-ai** [comment]
The provided information outlines a list of changes that have been made in the Elsa Workflows project between versions 3.1.2 and 3.2.0-rc1. These changes include various pull requests that have been merged into the project, addressing enhancements, bug fixes, new features, and performance improvements.

Some notable changes include:
- LogRecord storage at different levels.
- Minor improvements and bug fixes following the 3.1 release.
- Addition of settings for MassTransit message handling.
- Enhancements to BulkDispatchWorkflows Activity.
- Caching improvements to workflow runtime and workflow management stores.
- Refactoring of workflow dispatch and instance creation process.
- Updates to activity execution mapping in workflow runtime.
- Cache optimization for JavaScript Expressions.
- Removal of SerializedOutputValue from JournalData.
- Fixes for race conditions and null reference exceptions in various activities.
- Addition of distributed locking to MassTransitWorkflowDispatcher.
- JSON Serialization for Elsa expressions.
- MongoDB implementation for alteration storage.

Additionally, there are new contributors who have made their first contributions to the project.

For a full list of changes, you can view the full changelog on the GitHub release page for 3.2.0 RC1: https://github.com/elsa-workflows/elsa-core/releases/tag/3.2.0-rc1

This comment was generated by [Glime](https://glimelab.ai/?utm_source=dRbvTb7IhCSYiI&utm_medium=github).

**Comment by rboylesDev** [comment]
Excited about some of the improvements coming in this release! Initially testing with it has been good. Is there a timeframe for when this would get the full release?

**Reply by sfmskywalker** [reply]
Happy to hear that! Yes, release ETA is end of this week, early next week, depending on any last minute issues we may discover our receive as feedback.
